### PR TITLE
Configure Webpack/React (w/ babel)

### LINKED ISSUE #1: Configure Webpack/React (w/ babel for front end)

### PR DESCRIPTION
**Pull request by troym9731**
Closes #1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,13 +5,16 @@
   "private": true,
   "scripts": {
     "lint": "eslint . --cache",
-    "postinstall": "npm run build && npm prune",
+    "postinstall": "npm prune",
     "prestart": "npm run start:mongo",
-    "start": "nodemon --ignore build/ -e js,jsx --exec \"npm run build && nf start\"; npm stop",
+    "start": "scripts/start.sh",
```
Comment (therealklanni) [inline package.json]: `"start": "sh scripts/start/sh"` like I was saying in Slack. This is better than having to set modes.

File: src/client/scss/_animations.scss [hunk 2]
```diff
@@ -0,0 +1,114 @@
+// Home Page animation
```
Comment (therealklanni) [inline src/client/scss/_animations.scss]: Let's keep `client/` for JS files only. Can we move `scss/` up one level?

```
src
 |- client
 |- server
 |- scss (name it whatever you want)
```

Comment (troym9731) [inline src/client/scss/_animations.scss]: Absolutely. This PR is still a WIP

File: package.json [hunk 3]
```diff
@@ -5,16 +5,18 @@
...
     "lint": "eslint . --cache",
-    "postinstall": "npm run build && npm prune",
+    "postinstall": "npm prune",
     "prestart": "npm run start:mongo",
-    "start": "nodemon --ignore build/ -e js,jsx --exec \"npm run build && nf start\"; npm stop",
+    "start": "sh scripts/start.sh",
     "stop": "npm run stop:mongo",
     "start:mongo": "mongod --config mongod.conf --fork",
     "stop:mongo": "killall -9 mongod && echo Press ENTER to continue",
-    "build": "npm run build:clean && npm run build:server && npm run build:common",
```
Comment (therealklanni) [inline package.json:14]: Can we have `build` point to whatever we want built by default? dev or prod

File: package.json [hunk 4]
```diff
@@ -5,16 +5,18 @@
...
     "build:clean": "rimraf build",
     "build:common": "babel --source-root src --source-maps inline -d build/common src/common",
     "build:server": "babel --source-root src --source-maps inline -d build/server src/server",
+    "build:client": "sh scripts/build-client.sh",
+    "build:dev": "npm run build:server && npm run build:client",
```
Comment (therealklanni) [inline package.json]: `build:clean` should be run before any actual builds.

Comment (troym9731) [inline package.json]: Yeah, it's not set up correctly right now. I just took the lines from yours that will be used in the future

**Comment by therealklanni** [comment]
Looks good so far! :metal:

File: package.json (hunk 3 above)
Comment (troym9731) [inline package.json:14]: What would you consider default?

Comment (therealklanni) [inline package.json:14]: I usually make "dev" the default, because then us humans can type less ;)

File: package.json (hunk 4 above)
Comment (therealklanni) [inline package.json]: Gotcha :+1:

File: package.json (hunk 3 above)
Comment (troym9731) [inline package.json:14]: Sounds good, build will be dev by default

File: src/server/index.js [hunk 5]
```diff
@@ -39,12 +38,30 @@ import routes from './routes'
 
 app.use(routes)
 
-app.use(function *(next) {
-  this.body = 'Hello world'
-  yield next
-})
-
```
Comment (therealklanni) [inline src/server/index.js]: Should this maybe be protected by a `if !== 'production'` check?

File: src/server/index.js [hunk 6]
```diff
@@ -39,12 +38,30 @@ import routes from './routes'
...
 const port = process.env.PORT || 3000
 app.listen(port, () => {
   debug(`Server running at http://localhost:${port}`, perf.stop(serverStart))
 })
+
+// Webpack Dev Server for hot reloading
```
Comment (therealklanni) [inline src/server/index.js:46]: Should this maybe be protected by a `if !== 'production'` check?

File: src/server/index.js [hunk 7]
```diff
@@ -39,12 +38,30 @@ import routes from './routes'
...
+  stats: {
+    colors: true
+  },
+  devtool: 'inline-source-map'
+}).listen(3001, 'localhost', (err, result) => {
+  if (err) {
+    debug(err)
+  }
+
+  debug(`${result}: Listening at http://localhost:3001`)
```
Comment (therealklanni) [inline src/server/index.js]: Since I haven't really used this before, can you clue me in to what `result` is?

File: src/server/index.js (hunk 6 above)
Comment (troym9731) [inline src/server/index.js:46]: Yup

**Comment by therealklanni** [comment]
Looks really good, man! :metal:

File: src/server/index.js (hunk 7 above)
Comment (troym9731) [inline src/server/index.js]: In this case, undefined hahah. The docs have both err and result arguments, so I had both and used both.

**Pull request merged**